tests/providers/rust/kv-memory-bindgen/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+wasmcloud_provider_wit_bindgen::generate!(
+    // Impl Struct
+    KeyvalueProvider,
+    // WIT Namespace & package
+    "wasi:keyvalue",
+    // Interfaces that should be exposed on the lattice
+    ["wasi:keyvalue/readwrite", "wasi:keyvalue/atomic"],
+    // Everything that follows is fed to wasmtime component bindgen
+    "test-bindgen-kv-memory"
+);
+
+use async_trait::async_trait;
+use wasmcloud_provider_sdk::{core::LinkDefinition, error::ProviderResult, Context};
+
+use crate::wasi::keyvalue::types::{Bucket, Key, OutgoingValue};
+
+/// Implementation for wasmcloud:keyvalue
+#[derive(Default, Clone)]
+pub struct KeyvalueProvider;
+
+#[async_trait]
+impl WasiKeyvalueReadwrite for KeyvalueProvider {
+    // NOTE: these functions have to be filled out by hand instead of auto completed
+    // due to async_trait mucking things up.
+    //
+    // You'll get helpful compiler warnings about the methods being missing, but can't auto-complete
+    // the missing methods.
+
+    async fn get(
```

Comment:
If you insist on implementing keyvalue here, please use (tested) implementation in https://github.com/wasmCloud/wasmCloud/blob/28d2d6fc5e68ab8de12771fb3b0fb00617b32b30/crates/runtime/src/capability/provider/mem/keyvalue.rs, you can just import it. There is no need to maintain 2 separate implementations of the same thing

Reply:
Thanks, this is actually useful -- I can use that code instead of writing an implementation. 

As far I know, this memory implementation is not used by anyone outside this codebase, so *this* might be a good thing to port to bindgen first, since it is simpler, but I'm not going to do that in this ticket, to avoid unsettling any other tests.